### PR TITLE
CORE-13258 Avoid reaching out to database when there are no staterefs to resolve.

### PR DESCRIPTION
**Pull request by vlajos**
CORE-13258 Avoid reaching out to database when when there are no staterefs to resolve.

Performance tests: https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4505141266/Test+7+-+Reduce+time+spent+resolving+querying+stateRefs+within+ledger+flows+for+empty+inputs

### CODE REVIEW

**Review by lankydan — DISMISSED** [review]

**Review by blsemo — APPROVED** [review]
LGTM

**Review by mnesbit — APPROVED** [review]
Makes sense

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/UtxoLedgerStateQueryServiceImpl.kt [hunk 1]
```diff
@@ -37,26 +37,30 @@ class UtxoLedgerStateQueryServiceImpl @Activate constructor(
...
     @Suspendable
     override fun resolveStateRefs(stateRefs: Iterable<StateRef>): List<StateAndRef<*>> {
         return recordSuspendable({ ledgerPersistenceFlowTimer(ResolveStateRefs) }) @Suspendable {
-            wrapWithPersistenceException {
-                externalEventExecutor.execute(
-                    ResolveStateRefsExternalEventFactory::class.java,
-                    ResolveStateRefsParameters(stateRefs)
-                )
-            }.map { it.toStateAndRef<ContractState>(serializationService) }
+            if (stateRefs.count() == 0) {
```
Comment (mnesbit) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/UtxoLedgerStateQueryServiceImpl.kt:54]: My only concern here is that Iterable is not necessarily readable twice, so really API should be List<StateRef>, but it is almost certainly just a List in practice

Comment (mnesbit) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/persistence/UtxoLedgerStateQueryServiceImpl.kt:54]: Having said that do make damned sure that it does work when staterefs are non-empty

**Review by yarivus — APPROVED** [review]
LGTM

**Review by fenryka — APPROVED** [review]
LGTM

**Pull request merged**